src/main/java/org/wildfly/common/net/Inet.java

```diff
@@ -386,18 +385,47 @@ public static Inet4Address getInet4Address(int s1, int s2, int s3, int s4) {
         Assert.checkMaximumParameter("s3", 255, s3);
         Assert.checkMinimumParameter("s4", 0, s4);
         Assert.checkMaximumParameter("s4", 255, s4);
+        byte[] bytes = new byte[4];
         bytes[0] = (byte) s1;
         bytes[1] = (byte) s2;
         bytes[2] = (byte) s3;
         bytes[3] = (byte) s4;
+        // pre-compute the digits required
```

Comment:
You could avoid this (and some redundant branching) with something like:

```java
    byte[] hostBytes = new byte[15];
    int cnt = encodeUnsignedByte(s1, hostBytes, 0);
    hostBytes[cnt++] = '.';
    cnt += encodeUnsignedByte(s2, hostBytes, cnt);
    ...
    String hostName = new String(hostBytes, 0, cnt, StandardCharsets.ISO_8859_1);
```

and then:

```java
private static int encodeUnsignedByte(int value, byte[] bytes, int offset) {
    if (value >= 100) {
        bytes[offset++] = '0' + value / 100;
        value %= 100;
    }
    if (value >= 10) {
        ...
    }
    return offset;
}
```

Reply:
The redundant branching are there to avoid data dependency among `encodeUnsignedByte` which could be indipendent from each others, thanks to the upfront estimation. But is a minor: I'm opened to change it if you prefer it.

for

```java
String hostName = new String(hostBytes, 0, cnt, StandardCharsets.ISO_8859_1);
```
there's an opened JDK bug which show that it perform very bad, still, see https://bugs.openjdk.org/browse/JDK-8295496

quoting it

> I also included String(byte[], int, int, ISO_8859_1) which disappointingly lags slightly behind the deprecated String(byte[], int, int, int).

That's why I didn't used it.

Reply:
Additionally, I didn't used `cnt` and having properly sized byte arrays, exactly to simplify the life of the JDK stub generator and the bound checks, there, which proved to be the reason behind the bad performance of https://bugs.openjdk.org/browse/JDK-8295496